### PR TITLE
(#4871) - Switch to eslint

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+
+  "extends": "eslint:recommended",
+
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
```
Comment (marten-de-vries) [inline .eslintrc]: Why do we need jsx?

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  "globals": {
+    "chrome": true,
+    "Promise": true,
+    "Uint8Array": true,
+    "ArrayBuffer": true,
+    "FileReaderSync": true,
+    "sqlitePlugin": true,
+
+    "it": true,
```
Comment (marten-de-vries) [inline .eslintrc]: The mocha functions could be left out here by enabling the 'mocha' env (just like 'node' and 'browser')

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
```
Comment (marten-de-vries) [inline .eslintrc]: I think the convention is now to name the file .eslintrc.json instead of .eslintrc

**Comment by marten-de-vries** [comment]
Nice work looking into this. :)

**Comment by daleharvey** [comment]
Thanks for the comments / help @marten-de-vries, never knew about some of those, fixed them up and pushed a new PR

**Comment by daleharvey** [comment]
So one thing is this produces 453 warnings, almost entirely down to `no-unused-var` and `comma-dangle`. I would prefer to to have eslint fail on these which is why I set them to warn instead of ignore (the errors are almost entirely in the tests). I can follow up to remove all the warnings but wanted to get some input before doing that

**Comment by nolanlawson** [comment]
I'm +1 on removing dangling commas and unused vars, although I imagine most of those warnings are going to come from `tests` rather than the `src` or `bin`. Still, if someone is willing to make the fixes, +1 to fixing them everywhere.

**Comment by marten-de-vries** [comment]
If it's too much work to fix it everywhere at once it's also possible to selectively disable the warning in some files with a comment like:

```js
/* eslint comma-dangle: 0 */
```

**Comment by nickcolley** [comment]
:+1: for eslint, I can make the changes if you want @daleharvey

**Comment by daleharvey** [comment]
Didnt take a huge amount of time, I fixed all the warnings and turned them into errors :)

**Comment by daleharvey** [comment]
This will bitrot to hell and got a bunch of :+1: 's pluss all green so landing now, cheers 

https://github.com/pouchdb/pouchdb/commit/d2d339f722e365cb797dcd6d25f355db0e8f911b

**Pull request closed without merging**